src/enable_variable_ops/ngraph_assign_op.cc

```diff
@@ -0,0 +1,189 @@
+    Event event_compute(oss.str().c_str(), name().c_str());
+
+    NGRAPH_VLOG(4) << "NGraphAssign:: Compute called for: " << def().name()
+                   << " ,just looking " << just_looking_ << " ,copy-to-tf "
+                   << copy_to_tf_ << " ,Graph ID " << ng_graph_id_;
+
+    bool log_copies = false;
+    OP_REQUIRES_OK(context, IsCopyLogEnabled(ng_graph_id_, log_copies));
+    std::stringstream copy_log_str;
+    copy_log_str << "KERNEL[" << type_string() << "]: " << name()
+                 << " ,Copy_TF " << PrintBool(copy_to_tf_) << " ,Just_Looking "
+                 << PrintBool(just_looking_) << "\n";
+    int number_of_copies = 0;
+
+    bool ref_exists =
+        NGraphCatalog::ExistsInCatalog(ng_graph_id_, def().name(), 0);
+    if (!ref_exists) {
+      OP_REQUIRES(context, ref_exists,
+                  errors::Internal(
+                      "Caught exception : RefInput to NGAssign not found \n"));
+    }
+    string get_ref_var_name =
+        NGraphCatalog::GetInputSharedName(ng_graph_id_, def().name(), 0);
+
+    NGraphVar* var;
+    OP_REQUIRES_OK(context,
+                   context->resource_manager()->Lookup<NGraphVar>(
+                       context->resource_manager()->default_container(),
+                       get_ref_var_name, &var));
+
+    const Tensor& rhs = context->input(1);
+
+    // We always return the input ref.
+    context->forward_ref_input_to_ref_output(0, 0);
+
+    // get the nGraphTensor
+    shared_ptr<ngraph::runtime::Tensor> ng_tensor_to_assign = var->ng_tensor();
+
+    // DO NOT CARE ABOUT SYNCING AS WE ARE ALWAYS SETTING THE NGTENSOR
+
+    string valkey = to_string(ng_graph_id_) + "_" + def().input(1);
+    bool valref_exists = NGraphCatalog::ExistsInOutputCatalog(valkey);
+    if (valref_exists) {
+      // Value is from encap
+      NGRAPH_VLOG(4) << "NGraphAssign::Getting from catalog: " << valkey;
+      auto ng_val = NGraphCatalog::GetNgTensorFromOutputCatalog(valkey);
+      Event event_copy("D2D Copy", name().c_str());
+      ng_tensor_to_assign->copy_from(*ng_val);
+      event_copy.Stop();
+      Event::WriteTrace(event_copy);
+    } else {
+      number_of_copies++;
+      copy_log_str << " COPY_INP_VAL[0]";
+      NGRAPH_VLOG(4) << "NGraphAssign::Getting from TF : " << valkey;
+      void* tf_src_ptr = (void*)DMAHelper::base(&rhs);
+      Event event_host_2_dev_copy("H2D Copy", name().c_str());
+      ng_tensor_to_assign->write(
+          tf_src_ptr, 0, ng_tensor_to_assign->get_element_count() *
+                             ng_tensor_to_assign->get_element_type().size());
+      event_host_2_dev_copy.Stop();
+      Event::WriteTrace(event_host_2_dev_copy);
+    }
+
+    mutex_lock l(*context->input_ref_mutex(0));
+    Tensor old_lhs = context->mutable_input(0, /* lock_held */ true);
+
+    if (copy_to_tf_) {
+      number_of_copies++;
+      copy_log_str << " COPY_TF ";
+      ReadNGTensor(ng_tensor_to_assign, &old_lhs);
+
+      if (just_looking_) {
```

Comment:
empty `if`. We might rewrite it as `if (!just_looking_)`

Reply:
Done